test/e2e/cluster_extension_install_test.go

```diff
@@ -593,6 +598,90 @@ func TestClusterExtensionInstallReResolvesWhenManagedContentChanged(t *testing.T
 	}, pollDuration, pollInterval)
 }
 
+func TestClusterExtensionRecoversFromInitialInstallFailedWhenFailureFixed(t *testing.T) {
```

Comment:
Is this scenario testable in our reconcile unit tests yet? I'd love to avoid adding more e2es if possible.

Reply:
I'm not sure this test in particular is well suited to unit tests. In our unit tests we aren't necessarily testing the whole system. I imagine that we would still want an E2E test like this that flexes the entire system in this scenario.

I do think that separately we should go back through all the existing e2e tests and see if there are any that are focused in on a particular thing that could now be unit tested

Reply:
The reason I ask is that it seems like there is a set of preconditions that we could setup and an expected output. If we have scenarios that can be tested in a single execution of the `Reconcile` function, I think we should be trying to put those in unit tests.

Reply:
@everettraven and I chatted, and landed on making a separate epic that tracks improvements to make our code more testable at various layers.

We can merge with this e2e, and make testing/code structure improvements a separate task.